playbooks/migrating-postgres-with-rubyrep.md

```diff
@@ -0,0 +1,375 @@
+      config.options[:logged_replication_events] = :all_conflicts
+
+    end
+
+```
+
+Create the configmap with:
+```
+$ hokusai [staging|production] create --filename ./hokusai/rubyrep-config.yml
+```
+
+6) Create the file `./hokusai/rubyrep.yml`.  When the deployment is created, rubyrep will perform an initial sync.
+
+```
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: { myapp }-rubyrep
+  namespace: default
+spec:
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: { myapp }
+        component: rubyrep
+        layer: data
+      name: { myapp }-rubyrep
+    spec:
+      containers:
+        - name: rubyrep-{ myapp }
+          image: artsy/rubyrep
+          args: ["/rubyrep-2.0.1/rubyrep", "replicate", "-c", "/mnt/default.conf"]
+          imagePullPolicy: Always
+          resources:
+            requests:
+              cpu: 200m
+              memory: 256Mi
+          volumeMounts:
+            - name: rubyrep-{ myapp }
+              mountPath: /mnt/default.conf
+              subPath: default.conf
+      volumes:
+        - name: rubyrep-{ myapp }
+          configMap:
+            name: rubyrep-{ myapp }
+```
+
+Create the deployment with:
+```
+$ hokusai [staging|production] create --filename ./hokusai/rubyrep.yml
+```
+
+7) When the database is synced and replication is active, we are ready to make a cutover.
+
+  7a) Confirm that the databases are in-sync.  Run a rubyrep [scan command](http://www.rubyrep.org/scan_command.html) with:
+  ```
+  kubectl --context [staging|production] run rubyrep-$(whoami) --restart=Never --rm -i --tty --overrides '
+  {
+    "apiVersion": "v1",
+    "kind": "Pod",
+    "spec": {
+      "containers": [
+        {
+          "name": "rubyrep",
+          "image": "artsy/rubyrep",
+          "args": ["/rubyrep-2.0.1/rubyrep", "sync", "-c", "/mnt/default.conf", "-s"],
+          "stdin": true,
+          "stdinOnce": true,
+          "tty": true,
+          "volumeMounts": [{
+            "mountPath": "/mnt/default.conf",
+            "subPath": "default.conf",
+            "name": "rubyrep-{ myapp }"
+          }]
+        }
+      ],
+      "volumes": [{
+        "name": "rubyrep-{ myapp }",
+        "configMap": {
+          "name": "rubyrep-{ myapp }"
+        }
+      }]
+    }
+  }' --image artsy/rubyrep
+  ```
+
+  The diff should show `0` in total for all tables.
+
+  7b) To avoid race conditions regarding auto-incrementing primary keys, reset _all_ table id sequences except for `rr_*` RubyRep tables and id sequences such as `rr_pending_changes_id_seq`, leaving an offset for writes to the new database and headroom for replicated inserts from the old.
+
+  Run this command for _all_ tables / table_id_sequences in the database.  To get a list of all tables and id sequences, log into the database via `psql` and run `\d+`
+
+  ```
```

Comment:
Same here, perhaps a RubyRep command `rubyrep reset_id_sequences offset`?  My attempts to loop through all tables in PSQL while omitting all `rr_*` tables proved futile, but I think RubyRep already has biolderplat code to get an array of tables, excluding its own... it also used ActiveRecord so could then issue these composed statements 🤔